eslint.config.js

```diff
@@ -1,23 +1,60 @@
-import js from '@eslint/js'
-import globals from 'globals'
-import reactHooks from 'eslint-plugin-react-hooks'
-import reactRefresh from 'eslint-plugin-react-refresh'
-import tseslint from 'typescript-eslint'
-import { defineConfig, globalIgnores } from 'eslint/config'
+import js from '@eslint/js';
+import globals from 'globals';
+import reactHooks from 'eslint-plugin-react-hooks';
+import reactRefresh from 'eslint-plugin-react-refresh';
+import tseslint from 'typescript-eslint';
+import { defineConfig, globalIgnores } from 'eslint/config';
+
+import prettierPlugin from 'eslint-plugin-prettier';
+import simpleImportSortPlugin from 'eslint-plugin-simple-import-sort';
+import prettierConfig from 'eslint-config-prettier';
 
 export default defineConfig([
   globalIgnores(['dist']),
+
   {
     files: ['**/*.{ts,tsx}'],
     extends: [
       js.configs.recommended,
-      tseslint.configs.recommended,
+      ...tseslint.configs.recommended,
       reactHooks.configs.flat.recommended,
       reactRefresh.configs.vite,
     ],
     languageOptions: {
       ecmaVersion: 2020,
       globals: globals.browser,
     },
+
+    rules: {
+      '@typescript-eslint/no-unused-vars': [
+        'warn',
+        { argsIgnorePattern: '^_|^args$' },
+      ],
+
+      'react/prop-types': 'off',
+    },
+  },
+
+  {
+    files: ['**/*.{js,jsx,ts,tsx}'],
+
+    extends: [prettierConfig],
+
+    plugins: {
+      prettier: prettierPlugin,
+      'simple-import-sort': simpleImportSortPlugin,
+    },
+
+    rules: {
+      'prettier/prettier': ['error', { endOfLine: 'auto' }], // Prettier 규칙 위반 시 오류 처리
+      curly: ['error', 'all'], // 모든 제어문에 중괄호 {} 사용 강제
+      'no-console': 'warn', // console.log 경고
+      eqeqeq: ['error', 'always'], // === 사용 강제
+      'no-var': 'error', // var 사용 금지
```

Comment:
```suggestion
      'no-var': 'error',
      'prefer-const': 'off',
```
`no-var`만 있으니까 var, let으로 선언하면 모두 const로 되는 것 같아요 !
let은 사용할 가능성이 있을 수도 있으니까 `'prefer-const': 'off',` 이거 추가해주시면 좋을 것 같아요 ~!

Reply:
의도적으로 let을 사용하려고 작성한 부분도 아직 재할당이 없다는 이유로 자동으로 const로 변경되고 있더라고요. 그래서 개발자 입장에서는 오히려 흐름이 끊기거나 불편할 수도 있다고 느꼈어요.
prefer-const를 꺼주면 이런 자동 변경 없이 let을 그대로 유지할 수 있어서 그런 관점에서는 해당 옵션을 off 해주는 것도 좋겠다고 생각했어요.
물론 저희 프로젝트 특성상 let을 많이 사용할 일은 거의 없을 것 같아서 최종적으로는 진석님 판단에 따라주셔도좋을 것 같아요 ~!